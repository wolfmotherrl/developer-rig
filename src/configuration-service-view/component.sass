.configuration-service-view
  display: grid
  grid-template-columns: 1fr 8pt 1fr
  grid-gap: 8pt
  margin-top: 8pt

  &__button
    background-color: #6441A4
    border: 0
    color: white
    height: 15pt
    margin-left: 8pt
    min-width: 8em

    &--first
      margin-left: 0

  &__title
    font-size: larger
    font-weight: bold

  &-property
    display: block
    margin: 1em

    &__input
      width: 70%

      &--error
        border-color: red

    &__name
      margin-bottom: 5px
      margin-right: 11px

    &__value
      color: grey
      user-select: text

  &__vertical-bar
    background-color: #DAD8DE
    width: 1pt;
    height: 100%

  &__item
    margin-bottom: 8pt
    margin-left: -17pt

  &__item-text
    margin: 3pt

    &--title
      font-weight: bold
