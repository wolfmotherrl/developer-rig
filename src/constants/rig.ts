export const RigRole = 'rig';
export const DeveloperRigUserId = '265737932';

export enum LocalStorageKeys {
  CurrentProjectIndex = 'currentProjectIndex',
  ExtensionViews = 'extensionViews',
  Projects = 'projects',
  RigLogin = 'rigLogin',
}
