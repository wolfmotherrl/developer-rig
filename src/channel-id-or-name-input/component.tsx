import * as React from 'react';
import { DeveloperRigUserId } from '../constants/rig';

interface Props {
  isUser?: boolean;
  name: string;
  value: string;
  labelClassName?: string;
  textClassName?: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
}

interface State {
  checked: boolean;
  lastChannelId: string;
}


export class ChannelIdOrNameInput extends React.Component<Props, State> {
  state: State = {
    checked: this.props.value === DeveloperRigUserId,
    lastChannelId: '',
  }

  public render() {
    return (
      <>
        <label className={this.props.labelClassName}>
          <span className={this.props.textClassName}>
            {this.props.isUser ? 'User' : 'Channel'} ID or Name
          </span>
          <input type="text" name={this.props.name} value={this.props.value} onChange={this.props.onChange} disabled={this.state.checked} />
        </label>
        <label className={this.props.labelClassName}>
          <input type="checkbox" name="wantsDeveloperRigUserId" onChange={this.onChange} checked={this.state.checked} />
          Use Developer Rig user
        </label>
      </>
    );
  }

  private onChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { checked } = event.currentTarget;
    if (checked) {
      this.setState({ lastChannelId: this.props.value });
      this.props.onChange({ currentTarget: { name: this.props.name, value: DeveloperRigUserId } } as React.FormEvent<HTMLInputElement>);
    } else {
      this.props.onChange({ currentTarget: { name: this.props.name, value: this.state.lastChannelId } } as React.FormEvent<HTMLInputElement>);
    }
    this.setState({ checked });
  }
}
