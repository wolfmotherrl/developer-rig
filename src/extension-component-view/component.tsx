import * as React from 'react';
import { ExtensionFrame } from '../extension-frame';
import { FrameSize } from '../core/models/rig';
import { ExtensionMode, ExtensionViewType } from '../constants/extension-coordinator';
const { getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

interface ExtensionComponentViewProps {
  id: string;
  channelId: string;
  configuration: ExtensionCoordinator.Configuration;
  extension: ExtensionCoordinator.ExtensionObject;
  installationAbilities: ExtensionCoordinator.ExtensionInstallationAbilities;
  frameSize: FrameSize;
  position: ExtensionCoordinator.Position;
  role: string;
  isLocal: boolean;
  bindIframeToParent: (iframe: HTMLIFrameElement) => void;
}

type Props = ExtensionComponentViewProps & React.HTMLAttributes<HTMLDivElement>;

export class ExtensionComponentView extends React.Component<Props> {
  private computeViewStyles(): React.CSSProperties {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      extension.views.component);

    let viewStyles: React.CSSProperties = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        ...viewStyles,
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  public render() {
    return (
      <div
        className="view component-view nono_zone"
        style={{
          width: this.props.frameSize.width + 'px',
          height: this.props.frameSize.height + 'px',
        }}>
        <div style={this.computeViewStyles()}>
          <ExtensionFrame
            bindIframeToParent={this.props.bindIframeToParent}
            installationAbilities={this.props.installationAbilities}
            className="view"
            channelId={this.props.channelId}
            configuration={this.props.configuration}
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
            isLocal={this.props.isLocal}
            isPopout={false}
          />
        </div>
      </div>
    );
  }
}
