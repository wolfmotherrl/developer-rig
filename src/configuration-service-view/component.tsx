import * as React from 'react';
import './component.sass';
import { Configurations, RigProject } from '../core/models/rig';
import classNames = require('classnames');
import { fetchChannelConfigurationSegments } from '../util/api';
import { ChannelIdOrNameInput } from '../channel-id-or-name-input';
import { DeveloperRigUserId } from '../constants/rig';
import { fetchIdForUser } from '../util/id';

export interface Props {
  configurations: Configurations;
  rigProject: RigProject,
  userId: string;
  authToken: string;
  saveHandler: (segment: string, channelId: string, content: string, version: string) => void,
}

enum ConfigurationType {
  Broadcaster = 'broadcaster',
  Developer = 'developer',
  Global = 'global',
}

interface State {
  version: string;
  configurationType: ConfigurationType;
  channelId: string;
  configuration: string;
  fetchStatus: string;
  lastConfiguration: string;
  lastVersion: string;
  [key: string]: ConfigurationType | string;
}

export class ConfigurationServiceView extends React.Component<Props, State>{
  public state: State = {
    version: this.props.configurations.globalSegment && this.props.configurations.globalSegment.version || '',
    configurationType: ConfigurationType.Global,
    channelId: DeveloperRigUserId,
    configuration: this.props.configurations.globalSegment && this.props.configurations.globalSegment.content || '',
    fetchStatus: '',
    lastConfiguration: '',
    lastVersion: '',
  };

  private onChange = (input: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = input.currentTarget;
    this.setState({ [name]: value });
  }

  private onChangeConfigurationType = (input: React.FormEvent<HTMLSelectElement>) => {
    const { name, value } = input.currentTarget;
    this.setState({ [name]: value, lastConfiguration: '', lastVersion: '' });
    if (value === ConfigurationType.Global) {
      const { content = '', version = '' } = this.props.configurations.globalSegment || {};
      this.setState({ configuration: content, version });
    } else {
      this.setState({ configuration: '', version: '' });
    }
  }

  private fetchChannelConfiguration = async () => {
    let channelId = this.state.channelId.trim();
    if (channelId) {
      this.setState({ fetchStatus: 'fetching...' });
      try {
        channelId = await fetchIdForUser(this.props.authToken, channelId);
        let segmentMap = this.props.configurations.channelSegments[channelId];
        if (!segmentMap) {
          const { rigProject: { manifest: { id: clientId }, secret }, userId } = this.props;
          segmentMap = await fetchChannelConfigurationSegments(clientId, userId, channelId, secret);
        }
        const segment = this.state.configurationType === ConfigurationType.Broadcaster ?
          segmentMap.broadcaster : segmentMap.developer;
        const { content, version } = segment || { content: '', version: '' };
        this.setState({ configuration: content, lastConfiguration: content, version, lastVersion: version });
        this.setState({ fetchStatus: '' });
      } catch (ex) {
        this.setState({ fetchStatus: ex.message });
      }
    }
  }

  private canSave = (): boolean => {
    const { channelId, configurationType } = this.state;
    if (configurationType !== ConfigurationType.Global && !channelId.trim()) {
      // The non-global configuration segments need a channel ID.
      return false;
    }
    return true;
  }

  private save = async () => {
    if (this.canSave()) {
      const { configuration, configurationType, version } = this.state;
      let { channelId } = this.state;
      try {
        if (configurationType !== ConfigurationType.Global) {
          channelId = await fetchIdForUser(this.props.authToken, channelId);
        }
        this.props.saveHandler(configurationType, channelId, configuration.trim(), version.trim());
      } catch (ex) {
        this.setState({ fetchStatus: ex.message });
      }
    }
  }

  private cancel = () => {
    const { configurations } = this.props;
    const { content: configuration, version } = this.state.configurationType === ConfigurationType.Global ? configurations.globalSegment :
      { content: this.state.lastConfiguration, version: this.state.lastVersion };
    this.setState({ configuration, version });
  }

  private viewDocumentation() {
    window.open('https://dev.twitch.tv/docs/extensions/building/#configuration-service', 'developer-rig-help');
  }

  private viewTutorial() {
    window.open('https://www.twitch.tv/videos/320483709', 'developer-rig-help');
  }

  public render() {
    const versionClassName = classNames('configuration-service-view-property__input', {
      'configuration-service-view-property__input--error': !this.state.version.trim(),
    });
    const configurationClassName = classNames('configuration-service-view-property__input', {
      'configuration-service-view-property__input--error': !this.state.configuration.trim(),
    });
    const extensionUrl = `https://dev.twitch.tv/projects/${this.props.rigProject.manifest.id}/files`;
    return (
      <div className="configuration-service-view">
        <div className="configuration-service-view__section configuration-service-view__section--left">
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Configuration Type</div>
            <select className="configuration-service-view-property__select" name="configurationType" value={this.state.configurationType} onChange={this.onChangeConfigurationType}>
              <option value={ConfigurationType.Broadcaster}>Broadcaster</option>
              <option value={ConfigurationType.Developer}>Developer</option>
              <option value={ConfigurationType.Global}>Global</option>
            </select>
          </label>
          {this.state.configurationType !== ConfigurationType.Global && <>
            <ChannelIdOrNameInput name="channelId" value={this.state.channelId} onChange={this.onChange}
              labelClassName="configuration-service-view-property" textClassName="configuration-service-view-property__name" />
            <button className="configuration-service-view__button" onClick={this.fetchChannelConfiguration}>Fetch</button>
            <span>{this.state.fetchStatus}</span>
          </>}
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Configuration</div>
            <textarea className={configurationClassName} name="configuration" value={this.state.configuration} onChange={this.onChange} />
          </label>
          <label className="configuration-service-view-property">
            <div className="configuration-service-view-property__name">Version</div>
            <input className={versionClassName} type="text" name="version" value={this.state.version} onChange={this.onChange} />
          </label>
          <button className="configuration-service-view__button" onClick={this.save}>Save</button>
          <button className="configuration-service-view__button" onClick={this.cancel}>Cancel</button>
        </div>
        <div className="configuration-service-view__vertical-bar" />
        <div className="configuration-service-view__section configuration-service-view__section--right">
          <div className="configuration-service-view__title">Using the Configuration Service</div>
          <p className="configuration-service-view__text">The Configuration Service can be used in
            your extension by leveraging specific APIs in the Extension Helper Library.</p>
          <p className="configuration-service-view__text">To access, view, and directly edit your
            data in the Rig, pick Broadcaster, Developer, or Global, and if appropriate, search
            for a specific broadcaster who has installed your extension.  Alternatively, when
            testing your extension in the Extension Views tab, your extension front-end will be
            able to access the object returned by the Configuration Service on extension load.</p>
          <p className="configuration-service-view__text">Having Trouble?  Make sure youâ€™re set to
            use Hosted Configuration
            on <a href={extensionUrl} target="dev-site">your Extension Capabilities page</a>.
            Also you might need to refresh your Extension Manifest in the Rig.  You can do so on
            the Project Tab.  Happy Coding!</p>
          <button className="configuration-service-view__button configuration-service-view__button--first" onClick={this.viewTutorial}>View Tutorial</button>
          <button className="configuration-service-view__button" onClick={this.viewDocumentation}>Go to Documentation</button>
        </div>
      </div>
    );
  }
}
