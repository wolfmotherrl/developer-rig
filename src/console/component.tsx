import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  private console: React.RefObject<HTMLDivElement>;

  public state: State = {
    logHistory: window.rig.history || [],
  }

  constructor(props: {}) {
    super(props);
    this.console = React.createRef();
  }

  private updateConsole() {
    setTimeout(() => {
      const console = this.console.current;
      if (console) {
        const { offsetHeight, scrollHeight } = console;
        console.scrollTop = Math.max(0, scrollHeight - offsetHeight);
      }
    });
    this.setState({
      logHistory: window.rig.history
    });
  }

  public componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  public render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console" ref={this.console}>{logs}</div>
    );
  }
}
