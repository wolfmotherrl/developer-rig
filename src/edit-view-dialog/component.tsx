import * as React from 'react';
import * as closeButton from '../img/close_icon.png';
import './component.sass';
import { RadioOption } from '../extension-view-dialog/radio-option';
import { MobileOrientation } from '../constants/mobile';
import { RigExtensionView } from '../core/models/rig';
import { ExtensionViewType } from '../constants/extension-coordinator';

export interface EditViewProps {
  x?: number;
  y?: number;
  orientation?: string;
}

interface EditViewDialogProps {
  viewForEdit: RigExtensionView;
  closeHandler: () => void;
  saveViewHandler: (viewForEdit: RigExtensionView, newView: EditViewProps) => void;
}

interface State {
  x?: number;
  y?: number;
  orientation?: string;
  type?: string;
}

type Props = EditViewDialogProps;

export class EditViewDialog extends React.Component<Props, State> {
  public state: State = {
    x: this.props.viewForEdit.x,
    y: this.props.viewForEdit.y,
    orientation: this.props.viewForEdit.orientation,
    type: this.props.viewForEdit.type,
  }

  private renderOrientationComponents() {
    return Object.keys(MobileOrientation).map(option => {
      return <RadioOption
        key={option}
        name='orientation'
        value={MobileOrientation[option]}
        onChange={this.onChange}
        checked={option === this.state.orientation} />
    });
  }

  private onChange = (input: React.FormEvent<HTMLInputElement>) => {
    this.setState({
      [input.currentTarget.name]: input.currentTarget.value,
    });
  }

  private editClassFromType() {
    return this.state.type === ExtensionViewType.Component ? 'edit-view__dialog-component' :
      this.state.type === ExtensionViewType.Mobile ? 'edit-view__dialog-mobile' : 'edit-view__dialog';
  }

  public render() {
    return (
      <div className='edit-view'>
        <div className="edit-view__background" />
        <div className={this.editClassFromType()}>
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Edit View </div>
            <div className="top-bar-container__escape" onClick={this.props.closeHandler}><img alt="Close" src={closeButton} /></div>
          </div>
          <hr className="dialog__divider" />

          {this.state.type === ExtensionViewType.Component &&
            <div className="edit-view__content">
              <div className="type-and-size-container__type-title">
                Position (%)
              </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">X</label>
                    <input type="text" name="x" value={this.state.x} onChange={this.onChange} />
                  </div>
                  <div className="edit-subcontainer__input">
                    <label className="inputs__option-label">Y</label>
                    <input type="text" name="y" value={this.state.y} onChange={this.onChange} />
                  </div>
                </div>
              </div>
            </div>}

          {this.state.type === ExtensionViewType.Mobile &&
            <div className="size-title__size-subcontainer">
              <div className="size-subcontainer__presets">
                <div className="type-and-size-container__type-title">
                  Orientation
                </div>
                <div className='overlay-custom-container'>
                  {this.renderOrientationComponents()}
                </div>
              </div>
            </div>}

          <hr className="dialog__divider" />
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__save" onClick={() => this.props.saveViewHandler(this.props.viewForEdit, this.state)}> Save </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}
