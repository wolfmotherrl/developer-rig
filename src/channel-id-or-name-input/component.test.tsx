import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ChannelIdOrNameInput } from './component';
import { DeveloperRigUserId } from '../constants/rig';

describe('<ChannelIdOrNameInput />', () => {
  const setupShallow = setupShallowTest(ChannelIdOrNameInput, () => ({
    name: 'test',
    value: 'test',
    onChange: jest.fn(),
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('enables textbox', () => {
    const { wrapper } = setupShallow();
    wrapper.find('input[type="checkbox"]').simulate('change', { currentTarget: { checked: false } });
    expect(wrapper.instance().props.onChange).toHaveBeenCalledWith({ currentTarget: { name: "test", value: "" } });
  });

  it('disables textbox', () => {
    const { wrapper } = setupShallow();
    wrapper.find('input[type="checkbox"]').simulate('change', { currentTarget: { checked: true } });
    expect(wrapper.instance().props.onChange).toHaveBeenCalledWith({ currentTarget: { name: "test", value: DeveloperRigUserId } });
  });

  it('accepts input', () => {
    const { wrapper } = setupShallow();
    const event = { currentTarget: { value: 'test' } };
    wrapper.find('input[type="text"]').simulate('change', event);
    expect(wrapper.instance().props.onChange).toHaveBeenCalledWith(event);
  });
});
