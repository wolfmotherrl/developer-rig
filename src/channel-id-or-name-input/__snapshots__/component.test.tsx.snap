// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChannelIdOrNameInput /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChannelIdOrNameInput
    name="test"
    onChange={[Function]}
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          className={undefined}
>
          <span
                    className={undefined}
          >
                    Channel
                     ID or Name
          </span>
          <input
                    disabled={false}
                    name="test"
                    onChange={[Function]}
                    type="text"
                    value="test"
          />
</label>,
        <label
          className={undefined}
>
          <input
                    checked={false}
                    name="wantsDeveloperRigUserId"
                    onChange={[Function]}
                    type="checkbox"
          />
          Use Developer Rig user
</label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className={undefined}
>
              Channel
               ID or Name
</span>,
            <input
              disabled={false}
              name="test"
              onChange={[Function]}
              type="text"
              value="test"
/>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Channel",
                " ID or Name",
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "Channel",
              " ID or Name",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": false,
              "name": "test",
              "onChange": [Function],
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              name="wantsDeveloperRigUserId"
              onChange={[Function]}
              type="checkbox"
/>,
            "Use Developer Rig user",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "name": "wantsDeveloperRigUserId",
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Use Developer Rig user",
        ],
        "type": "label",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            className={undefined}
>
            <span
                        className={undefined}
            >
                        Channel
                         ID or Name
            </span>
            <input
                        disabled={false}
                        name="test"
                        onChange={[Function]}
                        type="text"
                        value="test"
            />
</label>,
          <label
            className={undefined}
>
            <input
                        checked={false}
                        name="wantsDeveloperRigUserId"
                        onChange={[Function]}
                        type="checkbox"
            />
            Use Developer Rig user
</label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className={undefined}
>
                Channel
                 ID or Name
</span>,
              <input
                disabled={false}
                name="test"
                onChange={[Function]}
                type="text"
                value="test"
/>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Channel",
                  " ID or Name",
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                "Channel",
                " ID or Name",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": false,
                "name": "test",
                "onChange": [Function],
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                name="wantsDeveloperRigUserId"
                onChange={[Function]}
                type="checkbox"
/>,
              "Use Developer Rig user",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "name": "wantsDeveloperRigUserId",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Use Developer Rig user",
          ],
          "type": "label",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
